crates/store/src/blocks.rs

```diff
@@ -0,0 +1,42 @@
+use std::{io::ErrorKind, path::PathBuf};
+
+#[derive(Debug)]
+pub struct BlockStore {
+    store_dir: PathBuf,
+}
+
+impl BlockStore {
+    pub async fn new(store_dir: PathBuf) -> Result<Self, std::io::Error> {
+        tokio::fs::create_dir_all(&store_dir).await?;
+
+        Ok(Self { store_dir })
+    }
+
+    pub fn save_block(&self, block_num: u32, data: &[u8]) -> Result<(), std::io::Error> {
+        let block_path = self.block_path(block_num);
+        let epoch_path = block_path.parent().ok_or(std::io::Error::from(ErrorKind::NotFound))?;
+
+        if !epoch_path.exists() {
+            std::fs::create_dir_all(epoch_path)?;
+        }
+
+        std::fs::write(block_path, data)
+    }
+
+    pub async fn load_block(&self, block_num: u32) -> Result<Option<Vec<u8>>, std::io::Error> {
+        match tokio::fs::read(self.block_path(block_num)).await {
+            Ok(data) => Ok(Some(data)),
+            Err(err) if err.kind() == std::io::ErrorKind::NotFound => Ok(None),
+            Err(err) => Err(err),
+        }
+    }
+
+    // HELPER FUNCTIONS
+    // --------------------------------------------------------------------------------------------
+
+    fn block_path(&self, block_num: u32) -> PathBuf {
+        let epoch = block_num >> 16;
+        let epoch_dir = self.store_dir.join(format!("{epoch:04x}"));
+        epoch_dir.join(format!("block_{block_num:08x}.dat"))
+    }
```

Comment:
I think this is fine for now, but I would still create an issue to discuss what should be the length of an epoch. For example, our plan is to reduce block times to 3 - 4 seconds, in which case, the current approach would mean that an epoch is just 2 - 3 days. This may be OK but also may be too short.

Reply:
If we use epoch only for limiting file count in the directory, I think, current epoch size is good. Even if we want to use it for something else, 2-3 days is still good (for example, in Solana each epoch lasts for about 2 days).

Reply:
Sounds good! I guess we can always introduce a higher-level periods that would be something like top 14 bits.

One potential change here: should epoch be formatted as: `{epoch:08x}` rather than `{epoch:04x}`?